### ISSUE TITLE
Cannot paste from Wayland clipboard after copying to it

### ISSUE THREAD

**Issue by iexos** [neovim/neovim#12063]
- `nvim --version`:
```
NVIM v0.4.3
Build type: Release
LuaJIT 2.0.5
Compilation: /usr/bin/cc -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -O2 -DNDEBUG -DMIN_LOG_LEVEL=3 -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fdiagnostics-color=always -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -I/build/neovim/src/build/config -I/build/neovim/src/neovim-0.4.3/src -I/usr/include -I/build/neovim/src/build/src/nvim/auto -I/build/neovim/src/build/include
Compiled by builduser

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info
```
- `vim -u DEFAULTS` (version: 0.4.3) behaves the same
- Operating system/version: Arch Linux 5.5.13-arch1-1 GNOME 3.36 on Wayland
- Terminal name/version: GNOME Terminal 3.36.0.1
- `$TERM`: screen-256color
- `wl-clipboard 2.0.0`

### Steps to reproduce using `nvim -u NORC`

```
# open two terminals. ">" denotes first terminal, "$" denotes second terminal
> nvim -u NORC
> it1<esc> # write line "t1"
$ nvim -u NORC
$ it2<esc> # write line "t2"
> "+yy   # copy "t1" into system clipboard
$ "+p    # paste system clipboard
$ k"+yy  # copy "t2" into system clipboard
> "+p    # try to paste system clipboard
> k"+yy  # copy "t1" into system clipboard
$ "+p    # try to paste system clipboard
```

### Actual behaviour
- second paste is line "t1"
- third paste is line "t2"

### Expected behaviour
- second paste is line "t2"
- third paste is line "t1"

Basically, after the first time I copy to `"+`, for the rest of the vim session it is not filled with the system clipboard again. This applies to copying from other sources as well. Copying from vim to the system clipboard still works fine.

**Comment by ainola** [comment]
FWIW, this does not affect me on Arch running sway. Your test scenario works as expected.

**Comment by glacambre** [comment]
I can't reproduce this on master, so I assume it has been fixed. @iexos let me know if you're still experiencing this on master/nightly and I'll re-open :).
